src/components/Card/Card.tsx

```diff
@@ -0,0 +1,61 @@
+import "./Card.scss";
+import type { Dessert } from "../../services/jsonService";
+import { useContext, useEffect, useReducer } from "react";
+import CardButton from "../CardButton/CardButton";
+import { CartContext } from "../../context";
+
+type Action = "add" | "substract" | "reset";
+
+function Card({ image, name, category, price }: Dessert) {
+  const updateCounter = (state: number, action: Action) => {
+    if (action === "add") {
+      return state + 1;
+    } else if (action === "substract"){
+    return state - 1;} else {
+      return 0
+    }
+  };
+  const [counter, setCounter] = useReducer(updateCounter, 0);
+  const cart = useContext(CartContext);
+
+  const handleClick = (action: Action) => {
+    setCounter(action);
+    console.log(counter);
+    if (action === "add") {
+      cart?.addItem({
+        name: name,
+        thumbnail: image.thumbnail,
+        quantity: counter+1,
+        price: price
+      });
+    } else {
+      cart?.addItem({
+        name: name,
+        thumbnail: image.thumbnail,
+        quantity: counter-1,
+        price: price
+      });
+    }
+  };
+
+  useEffect(() => {
+  const itemInCart = cart?.cart.find(item => item.name === name);
+  if (!itemInCart) {
+    setCounter('reset'); // reset if removed
+  } 
+}, [cart?.cart, name, counter]);
+  return (
+    <div className="card">
+      <img className="card__image" src={image.desktop} alt="" />
```

Comment:
```suggestion
      <img className="card__image" src={image.desktop} alt="" aria-hidden='true' />
```
this is just a suggestion to avoid losing SEO, an empty alt on an image is always punished on SEO.

Reply:
Actually I had no idea the alts affected SEO, so that's a good reason to remind myself to manage them properly